PaleTree.Shared/Op.cs

```diff
@@ -1,2226 +1,43 @@
-		public const int ZC_SET_AURA_INFO = 0xFF8; // Size: 275
-		public const int CZ_OPEN_HELP = 0xFF9; // Size: 86
-		public const int CZ_REQ_UNKNOWN_SANTUARTY_BUFF = 0xFFA; // Size: 346
-		public const int ZC_SEND_CURRENT_JOIN_MGAME_NAME = 0xFFB; // Size: 266
-		public const int CZ_ANS_GIVEUP_PREV_PLAYING_INDUN_PARTY = 0xFFC; // Size: 23
-		public const int CS_LOGIN = 0x3E1D; // Size: 347
-		public const int SC_NORMAL = 0x3E1E; // Size: 0
-		public const int SC_FROM_INTEGRATE = 0x3E1F; // Size: 0
-		public const int CS_REQ_ADD_FRIEND = 0x3E20; // Size: 74
-		public const int CS_REQ_BLOCK_FRIEND = 0x3E21; // Size: 74
-		public const int CS_FRIEND_CMD = 0x3E22; // Size: 32
-		public const int CS_FRIEND_SET_ADDINFO = 0x3E23; // Size: 168
-		public const int CS_CHAT = 0x3E24; // Size: 0
-		public const int CS_GROUP_CHAT_INVITE = 0x3E25; // Size: 114
-		public const int CS_GROUP_CHAT_INVITE_BY_TAG = 0x3E26; // Size: 26
-		public const int CS_ALLOW_GROUP_CHAT_TAG_INVITE = 0x3E27; // Size: 50
-		public const int CS_REFRESH_GROUP_CHAT = 0x3E28; // Size: 10
-		public const int CS_CREATE_GROUP_CHAT = 0x3E29; // Size: 10
-		public const int CS_REQ_CHAT_HISTORY = 0x3E2A; // Size: 26
-		public const int CS_REQ_OUT_ROOM = 0x3E2B; // Size: 18
-		public const int CS_REQ_RELATED_PC_SESSION = 0x3E2C; // Size: 24
-		public const int CS_REDIS_SKILLPOINT = 0x3E2D; // Size: 26
-		public const int CS_PARTY_CLIENT_INFO_SEND = 0x3E2E; // Size: 0
-		public const int CS_NORMAL_GAME_START = 0x3E2F; // Size: 10
-		public const int CS_REQUEST_PVP_RANKING = 0x3E30; // Size: 92
-		public const int CS_REQUEST_ALL_SEASON_TOP_PVP_RANKING = 0x3E31; // Size: 84
-		public const int CS_INVITE_PARTY_PVP = 0x3E32; // Size: 14
-		public const int CS_ACCEPT_PARTY_PVP = 0x3E33; // Size: 23
-		public const int CS_LIKE_IT = 0x3E34; // Size: 96
-		public const int CS_UNLIKE_IT = 0x3E35; // Size: 32
-		public const int CS_LIKE_IT_CONFIRM = 0x3E36; // Size: 24
-		public const int CS_ADD_RELATION_SCORE = 0x3E37; // Size: 96
-		public const int CS_GET_LIKE_COUNT = 0x3E38; // Size: 24
-		public const int CS_REQ_MEMBER_LOGOUT_TIME = 0x3E39; // Size: 10
-		public const int CS_INVITE_GROUPCHAT_ACCEPT = 0x3E3A; // Size: 135
-		public const int SC_LOGIN_OK = 0x3E3B; // Size: 10
+    public static class Op
+    {
+        static readonly Dictionary<int, int> OPSize = new Dictionary<int, int>();
+        static readonly Dictionary<int, string> OPName = new Dictionary<int, string>();
 
-		private static readonly Dictionary<int, int> _sizes = new Dictionary<int, int>();
-		private static readonly Dictionary<int, string> _names = new Dictionary<int, string>();
+        public static void FillDictionary()
+        {
+            using (StreamReader reader = new StreamReader("Zemyna_Ops.txt"))
+            {
+                string line;
+                while ((line = reader.ReadLine()) != null)
+                {
+                    string[] parts = line.Split(' ');
```

Comment:
You're making several assumptions about the format of the op list here. If someone wanted to update or modify the list, and they didn't get it exactly right, like using a space more or less, your code would immediately fail and throw exceptions. What would you say about using regular expressions here? They might make the parsing easier to follow and more robust. For example:

```cs
var match = Regex.Match(line, @"(?<name>.+?)\s*:\s*(?<op>.+?)\s*\(\s*(?<size>.+?)\s*\)");
if (!match.Success)
    continue;

var name = match.Groups["name"].Value.Trim();
var op = match.Groups["op"].Value.Trim();
var size = match.Groups["size"].Value.Trim();
```

You might also want to use `int.TryParse` and give appropriate error messages if something goes wrong, because right now, the user would just get the default exception dialog, that's really not all that useful to them.

Reply:
I'll implement it, even though database structures should pretty much come as they are. If the user adds an additional delimiter and the program breaks, that's mostly on them. Try importing a csv file to an sql db with a duplicated delimiter.
Btw any reason as to why Zemyna is using such a weird structure? string: hex (int), any reason for those brackets and the colon?

Reply:
In my eyes it's not a database, it's a list, in a human readable and editable format. Think of it less as -mechanical voice on- "value;value;value", and more like "the op with this name has: this value (and the size is this)". Simply feels more natural to me, when not viewed through a programmer's eyes.